pkg/v1/remote/transport/transport.go

```diff
@@ -96,8 +96,19 @@ func NewWithContext(ctx context.Context, reg name.Registry, auth authn.Authentic
 		if err := bt.refresh(ctx); err != nil {
 			return nil, err
 		}
-		return bt, nil
+		return &Wrapper{bt}, nil
 	default:
 		return nil, fmt.Errorf("unrecognized challenge: %s", pr.challenge)
 	}
 }
+
+// Wrapper results in *not* wrapping supplied transport with additional logic such as retries, useragent and debug logging
+// Consumers are opt-ing into providing their own transport without any additional wrapping.
+type Wrapper struct {
+	Inner http.RoundTripper
```

Comment:
Let's not expose `Inner` quite yet? I think we only want you to be able to produce a `Wrapper` by going through `transport.New` for now. We can relax that later if it makes sense.

Reply:
ok done!